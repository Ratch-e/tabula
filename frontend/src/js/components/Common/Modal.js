import * as React from "react";
import * as classnames from "classnames"
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import * as personsAction from '../../actions/PersonsActions'

class Modal extends React.Component {
  constructor(props) {
    super(props);

    this.deletePerson = this.deletePerson.bind(this);
  }

  deletePerson() {
    this.props.personActions.removePerson(this.getId());
  }

  getId() {
    return this.props.id;
  }

  render() {
    return (
      <div className={classnames('modal',{'modal_hidden': !this.props.active})}>
        <div className="modal__overlay" onClick={this.props.toggle}></div>
        <div className="modal__content">
          <div className="modal__close" onClick={this.props.toggle}>X</div>
          <div className="modal__text">Вы уверены что хотите удалить эту анкету? Это действие необратимо!</div>
          <div className="modal__options">
            <div className="button" onClick={this.props.toggle}>Отмена</div>
            <div className="button button_warning" onClick={this.deletePerson}>Удалить</div>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps (state) {
  return {
    persons: state.persons
  }
}

function mapDispatchToProps(dispatch) {
  return {
    personActions: bindActionCreators(personsAction, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Modal);