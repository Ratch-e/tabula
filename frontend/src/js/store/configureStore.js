import {createStore, applyMiddleware} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index'


export default function configureStore(initialState) {

  const store = createStore(rootReducer, initialState, composeWithDevTools(
    applyMiddleware(thunk)
  ));

  if (module.hot) {
    module.hot.accept('../reducers', () => {
      const nextRootReducer = require('../reducers/index');
      store.replaceReducer(nextRootReducer)
    });
  }

  return store;
}