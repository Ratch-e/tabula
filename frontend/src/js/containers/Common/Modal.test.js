import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Modal from './Modal';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Модальное окно', () => {
  it('рендерится корректно', () => {
    const initialState = {};
    const store = mockStore(initialState);
    const component = shallow(
      <Provider store={store}>
        <Modal />
      </Provider>,
    )
      .dive({ context: { store } })
      .dive();
  
    expect(component.length).toMatchSnapshot();
  });
});
