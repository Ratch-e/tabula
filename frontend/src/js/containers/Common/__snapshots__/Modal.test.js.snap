// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Модальное окно рендерится корректно 1`] = `1`;
